### DISCUSSION TITLE
why the 'delelte from diagram' button is not working.

### DISCUSSION THREAD

**Discussion by Willia-Yang** [eclipse-sirius/sirius-desktop#487] [Q&A]
I don't why the 'delelte from diagram' button is not working, but  'delelte from Model' button is working, I want they are all working.
![image](https://github.com/user-attachments/assets/d6c0ad8e-5951-4bbd-b952-e3053bb7d50d)

**Comment by lredor** [comment]
Hi,
As said in the user documentation: `Delete from Diagram. This action removes the selected graphical element from the diagram, but does not delete the corresponding semantic elements. It is only available on un-synchronized diagrams or element types.`
You can have a look to [this documentation](https://eclipse.dev/sirius/doc/user/diagrams/Diagrams.html#synchronized_diagram) about unsynchronized diagrams and to `Synchronization` section of [this documentation](https://eclipse.dev/sirius/doc/specifier/diagrams/Diagrams.html#graphical_elements) for more explanation about unsynchronized elements.
Regards,
Laurent

**Reply by Willia-Yang** [reply]
I already know the difference between these two kinds of deletion, I want to know if it is possible to have both deletion functions for one element.

**Reply by lredor** [reply]
To be more clear, the `delete from diagram` is not available probably because your element and/or your diagram is not unsynchronized.

**Comment by Willia-Yang** [comment]
Thank you for your apply, i will have a look to the documentation.
